src/units/abilities.cpp

```diff
@@ -600,28 +614,31 @@ namespace {
 bool attack_type::get_special_bool(const std::string& special, bool simple_check) const
 {
 	{
-		std::vector<const config*> list;
-		if ( get_special_children(list, specials_, special, simple_check) ) {
+		std::vector<special_match> special_tag_matches;
+		std::vector<special_match> special_id_matches;
+		if ( get_special_children(special_tag_matches, special_id_matches, specials_, special, simple_check) ) {
 			return true;
 		}
 		// If we make it to here, then either list.empty() or !simple_check.
 		// So if the list is not empty, then this is not a simple check and
 		// we need to check each special in the list to see if any are active.
-		for(const config* entry : list) {
-			if ( special_active(*entry, AFFECT_SELF) ) {
+		for(const special_match entry : special_tag_matches) {
+			if ( special_active(entry.cfg, AFFECT_SELF, entry.tag_name) ) {
 				return true;
 			}
 		}
```

Comment:
It's possible that I'm missing something simple, but shouldn't there be another loop over `special_id_matches`? get_special_bool is (currently) documented to return true if _either_ tag or id matches, but as the PR stands it won't return true for id matches, only for tag name matches?